tests/scopeguard.cpp

```diff
@@ -0,0 +1,79 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "scopeguard.h"
+
+#include <exception>
+
+namespace bpftrace {
+
+TEST(ScopeGuardTest, InnerScope)
+{
+  int x = 5;
+
+  {
+    SCOPE_EXIT
+    {
+      x++;
+    };
+  }
+
+  EXPECT_EQ(x, 6);
+}
+
+TEST(ScopeGuardTest, FunctionReturn)
+{
+  int x = 5;
+
+  [&]() {
+    SCOPE_EXIT
+    {
+      x++;
+    };
+
+    x++;
+    return;
+  }();
+
+  EXPECT_EQ(x, 7);
+}
+
+TEST(ScopeGuardTest, ExceptionContext)
+{
+  int x = 5;
+
+  try {
+    [&]() {
+      SCOPE_EXIT
+      {
+        x++;
+      };
+
+      throw std::runtime_error("exception");
+    }();
+  } catch (const std::exception &) {
```

Comment:
super nit. I might add an ordering part of the test here where you expect x to be 6 and then increment it and expect x to be 7 below.